### PR TITLE
Dev3005

### PR DESCRIPTION
**Pull request by hfurkanbozkurt**

### CODE REVIEW

File: doc/source/installation.rst [hunk 1]
```diff
@@ -1,9 +1,116 @@
 .. _installation:
 
+Requirements
+============
+
+PCG has the following system requirements:
+
+- Linux based operating system
+- Python (>=3.5)
+- C++ compiler (with C++11 supports) and SWIG (version 3.0 or later)
```
Comment (AndreBiedenkapp) [inline doc/source/installation.rst]: why are we dependent on SWIG?

File: pcg/envs/clingo_env.py [hunk 2]
```diff
@@ -342,3 +348,22 @@ def action_fun(action, prg):  # Default action function assumes we control resta
...
             self.action_fun = action_fun
             self.reward_fun = reward_fun
+
+    def set_logger(self, log_to_std=False, log_to_txt=True):
+        self.logger = logging.getLogger(
+            self.__module__ + "." + self.__class__.__name__)
+
+        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+        if True:
```
Comment (AndreBiedenkapp) [inline pcg/envs/clingo_env.py]: I believe the query should be
```python
if log_to_std:
```

File: pcg/envs/clingo_env.py [hunk 3]
```diff
@@ -342,3 +348,22 @@ def action_fun(action, prg):  # Default action function assumes we control resta
         else:
             self.action_fun = action_fun
             self.reward_fun = reward_fun
+
+    def set_logger(self, log_to_std=False, log_to_txt=True):
+        self.logger = logging.getLogger(
```
Comment (AndreBiedenkapp) [inline pcg/envs/clingo_env.py]: I'm not 100% certain as to how 
```python
self.__class__
```
works for inheritance but
This method could probably live in the base class.

File: pcg/envs/saps_env.py [hunk 4]
```diff
@@ -143,3 +140,30 @@ def reset(self):
             if not line:
                 time.sleep(0.1)
                 continue
+
+
+    def shape_env(self, **kwargs):
+        pass
```
Comment (AndreBiedenkapp) [inline pcg/envs/saps_env.py]: A `raise NotImplementedError` might be more fail-safe as a potential user will get immediate feedback that this method does not work (yet)

File: pcg/envs/saps_env.py [hunk 5]
```diff
@@ -143,3 +140,30 @@ def reset(self):
...
+
+    def close(self):
+        self.logger.info("Closing the environment.")
+
+    def render(self, mode='human', close=True):
+        pass
```
Comment (AndreBiedenkapp) [inline pcg/envs/saps_env.py]: Same comment as for shape_env

**Comment by AndreBiedenkapp** [comment]
@hfurkanbozkurt Thanks to the autobuild of the documentation we can now publish our docs at
https://automl.github.io/gym-parameter-control

File: pcg/envs/clingo_env.py (hunk 3 above)
Comment (hfurkanbozkurt) [reply pcg/envs/clingo_env.py]: @AndreBiedenkapp Should we change it?

**Pull request merged**